azurelinuxagent/common/protocol/wire.py

```diff
@@ -510,10 +510,8 @@ def event_to_v1(event):
     params = ""
     for param in event.parameters:
         params += event_param_to_v1(param)
-    event_str = ('<Event id="{0}">'
-                 '<![CDATA[{1}]]>'
-                 '</Event>').format(event.eventId, params)
-    return event_str
+    event_str = ustr('<Event id="{0}"><![CDATA[{1}]]></Event>').format(event.eventId, params)
+    return event_str.encode("utf-8")
```

Comment:
Wouldn't we want to keep event_str as a ustr object? That way, we can limit encoding (and therefore, raw strings that will except on calls to formats with unicode args) to writing to disk and telemetry.

Reply:
This is actually the step before sending it out to telemetry. After this we directly call the common restutil function which actually sends it out but I didn't want to change it there because that is being called from a lot of places and that introduces a bigger scope for an error.

Reply:
can we add utf_8 somewhere in the function name to be more explicit on what it returns ?